### PR TITLE
Blog authors

### LINKED ISSUE #5023: [Bug] Blog author text no longer shows on the page
**This is currently happening on the new homepage branch, and not production.**

I'm guessing this is due to the recent blog (author) model change and that we forgot to update the template code to reflect that.

### Case 1. On blog index page. (`/blog`)

image

Author info is missing in the highlighted line. It should read `By [author] on Aug. 4, 2020`

### Case 2. On blog post page.

image

It should read `By [author] | Aug. 4, 2020 | Advocacy`

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #5023

- homepage (please focus on the "spotlight" section as the rest of page are being worked on in other PRs)
- blog index
- blog post

@beccaklam I'm assigning you as design reviewer as it seems like this design work was primarily done by you. I could be wrong so feel free to re-assign this to another designer!

I also wanna confirm that we no longer wanna show `category` info on the blog post page since it's not on the mockups anymore.

### CODE REVIEW

**Comment by mmmavis** [comment]
@beccaklam thanks for the feedback. I have adjusted the overlap spacing for author image and updated the line-height so it's 18px instead.

- I can add the `category` info back. Could you update the Sketch with it (along with the style for link hover state)?

- Regarding turning the blog index page into one column on mobile -
Yup, it's totally possible. But we should tackle it in a different ticket as I think that work might touch other index page code and that this PR is solely to get the blog author stuff updated!

**Comment by beccaklam** [comment]
> @beccaklam thanks for the feedback. I have adjusted the overlap spacing for author image and updated the line-height so it's 18px instead.
> 
>     * I can add the `category` info back. Could you update the Sketch with it (along with the style for link hover state)?
> 
>     * Regarding turning the blog index page into one column on mobile(? or it is for tablet?) -
>       Yup, it's totally possible. But we should tackle it in a different ticket as I think that work might touch other index page code and that this PR is solely to get the blog author stuff updated! 👍

Yup, updated Sketch file so the Category is included in the blog post and included the link hover state.
Yeah that's fine about covering the one column change in another ticket!

**Comment by beccaklam** [comment]
@mmmavis The changes you made also look great! Thank you! I just saw one thing -- for the blog post, would it be more consistent if the author names followed the same rules on the other pages where it starts on a new line if they _can't_ all fit on the same line but are beside the photos if the names _can_ all fit on the same line? Let me know if that makes sense!
Screen Shot 2020-08-13 at 5 51 59 PM

**Comment by mmmavis** [comment]
@beccaklam 
> for the blog post, would it be more consistent if the author names followed the same rules on the other pages where it starts on a new line if they can't all fit on the same line but are beside the photos if the names can all fit on the same line? Let me know if that makes sense!

Either way sounds fine to me!

**Comment by beccaklam** [comment]
@mmmavis Okay, let's just follow what we do on the other pages. So names besides the photos when all the names will fit, and start on the next line when they won't.

**Comment by mmmavis** [comment]
> @mmmavis Okay, let's just follow what we do on the other pages. So names besides the photos when all the names will fit, and start on the next line when they won't.

Sorry @beccaklam . It seems like Sketch file hasn't been updated with the design changes. Because more back and forth convo to deal with design changes will take more time and that we want to wrap up our homepage work hopefully by tomorrow, let's get this PR reviewed based on the original design. Please file a ticket for the new design change so we can tackle it after homepage work is live.

File: network-api/networkapi/wagtailpages/factory/blog.py [hunk 1]
```diff
@@ -51,6 +53,17 @@ def add_category(post):
     post.save()
 
 
+def add_authors(post):
+    authors = list(BlogAuthor.objects.order_by("?").all())
+    count = len(authors)
+
+    for i in range(0, randint(1, min(count, 5))):
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/factory/blog.py:60]: Nice randomization!

**Comment by KalobTaulien** [comment]
Looks good @mmmavis, nice work!

**Pull request merged**